CampaignConfig/mdc2020_mix.cfg

```diff
@@ -0,0 +1,268 @@
+executable_2.arg_4 = --inputs=inputs.txt
+executable_2.arg_5 = --merge-factor=1
+executable_2.arg_6 = --embed
+executable_2.arg_7 = template.fcl
+
+job_setup.ifdh_art = False
+
+[stage_ceendpointdigi]
+global.stage_name = CeEndpointDigi
+job_output_1.dest = %(logdir_bck)s/%(stage_name)s/%(desc)s/tbz/
+
+job_output_2.addoutput = dig.*.CeEndpointTriggered.*art
+job_output_2.add_to_dataset = dig.%(submitter)s.CeEndpointTriggered.%(desc)s.art
+job_output_2.dest = %(outdir_dig_tape)s/CeEndpointTriggered/%(desc)s/art
+
+job_output_3.addoutput = dig.*.CeEndpointUntriggered.*art
+job_output_3.add_to_dataset = dig.%(submitter)s.CeEndpointUntriggered.%(desc)s.art
+job_output_3.dest = %(outdir_dig_tape)s/CeEndpointUntriggered/%(desc)s/art
+global.artRoot_dataset = dig.%(submitter)s.CeEndpointTriggered.%(desc)s.art,dig.%(submitter)s.CeEndpointUntriggered.%(desc)s.art
+
+submit.dataset = cnf.%(submitter)s.%(stage_name)s.%(desc)s.fcl
+submit.n_files_per_job = 1
+sam_consumer.limit = 1
+job_setup.postscript = samweb file-lineage parents `basename ${fname}` > parents.txt
+
+job_setup.getconfig = True
+job_setup.multifile = False
+job_setup.setup_local = True
+
+[stage_ceendpointmix_fcl]
+global.stage_name = CeEndpointMix
+job_output.dest = %(outdir_fcl)s/%(stage_name)s/%(desc)s/fcl/
+
+job_setup.prescript = samweb list-file-locations --defname=dts.%(submitter)s.CeEndpoint.%(desc)s.art --schema=root | cut -f1 > CeEndpoint%(release_v)s.txt
+
+executable.name = true
+executable_2.name = gen_Mix.sh
+executable_2.arg_1 = CeEndpoint
+executable_2.arg_2 = %(release_v)s
+executable_2.arg_3 = %(njobs)s
+executable_2.arg_4 = 2000
+
+job_setup.ifdh_art = False
+
+[stage_ceendpointmix]
+global.stage_name = CeEndpointMix
+job_output_1.dest = %(logdir_bck)s/%(stage_name)s/%(desc)s/tbz/
+
+job_output_2.addoutput = dig.*.CeEndpointTriggeredMix.*art
+job_output_2.add_to_dataset = dig.%(submitter)s.CeEndpointTriggeredMix.%(desc)s.art
+job_output_2.dest = %(outdir_dig_tape)s/CeEndpointTriggeredMix/%(desc)s/art
+
+job_output_3.addoutput = dig.*.CeEndpointUntriggeredMix.*art
+job_output_3.add_to_dataset = dig.%(submitter)s.CeEndpointUntriggeredMix.%(desc)s.art
+job_output_3.dest = %(outdir_dig_tape)s/CeEndpointUntriggeredMix/%(desc)s/art
+global.artRoot_dataset = dig.%(submitter)s.CeEndpointTriggeredMix.%(desc)s.art,dig.%(submitter)s.CeEndpointUntriggeredMix.%(desc)s.art
+
+submit.dataset = cnf.%(submitter)s.%(stage_name)s.%(desc)s.fcl
+submit.n_files_per_job = 1
+sam_consumer.limit = 1
+job_setup.postscript = samweb file-lineage parents `basename ${fname}` > parents.txt
+
+job_setup.getconfig = True
+job_setup.multifile = False
+job_setup.setup_local = True
+
+[stage_noprimarymix_fcl]
+global.stage_name = NoPrimaryMix
+job_output.dest = %(outdir_fcl)s/%(stage_name)s/%(desc)s/fcl/
+
+executable.name = true
+executable_2.name = gen_Mix.sh
+executable_2.arg_1 = NoPrimary
+executable_2.arg_2 = %(release_v)s
+executable_2.arg_3 = %(njobs)s
+executable_2.arg_4 = 2000
```

Comment:
Is art_4 the event count?  Mixing jobs should process an entire file.  Also, isn't njobs set by the size of the input collection?  Does it have to be manually set here too?

Reply:
This is for the NoPrimary, so there is no input collection. I took those numbers from your version of gen_Mix.sh https://github.com/Mu2e/Production/blob/7528da7d0ac43bc4c332d3fe56312a36294e66e5/MDC2020/gen_Mix.sh#L63